hr_holidays_accrual/models/hr_leave_allocation.py

```diff
@@ -0,0 +1,412 @@
+# Copyright (C) 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from math import ceil
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+from odoo import models, fields, api, _
+from odoo.addons.resource.models.resource import HOURS_PER_DAY
+
+_logger = logging.getLogger(__name__)
+
+
+class HrLeaveAllocation(models.Model):
+    _inherit = 'hr.leave.allocation'
+
+    accruement_ids = fields.One2many(
+        string='Accruements',
+        comodel_name='hr.leave.allocation.accruement',
+        inverse_name='leave_allocation_id',
+        readonly=True,
+    )
+    limit_accrued_days = fields.Boolean(
+        'Limit Number of Days Accrued',
+        track_visibility='onchange',
+        readonly=True,
+        states={
+            'draft': [('readonly', False)],
+            'confirm': [('readonly', False)]
+        },
+        help=(
+            'Limit total maximum number of days accrued in one accrual'
+            ' period.'
+        ),
+    )
+    max_accrued_days = fields.Float(
+        'Max Number of Days Accrued',
+        track_visibility='onchange',
+        readonly=True,
+        states={
+            'draft': [('readonly', False)],
+            'confirm': [('readonly', False)]
+        },
+        help='Total maximum number of days accrued in one accrual period.',
+    )
+    limit_carryover_days = fields.Boolean(
+        'Limit Number of Days to Carryover',
+        track_visibility='onchange',
+        readonly=True,
+        states={
+            'draft': [('readonly', False)],
+            'confirm': [('readonly', False)]
+        },
+        help=(
+            'Limit total maximum number of days to carryover to next accrual'
+            ' period.'
+        ),
+    )
+    max_carryover_days = fields.Float(
+        'Max Number of Days to Carryover',
+        track_visibility='onchange',
+        readonly=True,
+        states={
+            'draft': [('readonly', False)],
+            'confirm': [('readonly', False)]
+        },
+        help=(
+            'Total maximum number of days to carryover to next accrual period.'
+        ),
+    )
+    limit_accumulated_days = fields.Boolean(
+        'Limit Total Balance',
+        track_visibility='onchange',
+        readonly=True,
+        states={
+            'draft': [('readonly', False)],
+            'confirm': [('readonly', False)]
+        },
+        help='Limit total maximum number of days that can be accumulated.',
+    )
+    max_accumulated_days = fields.Float(
+        'Total Balance Limit',
```

Comment:
This string is probably wrong considering the field above.

Reply:
Probably it's confusing, one is "value" other is flag "do apply". In UI it's more clear "[Checkbox] Limit Total Balance" and "Total Balance Limit: [Editbox]"